Formula/a/apache-spark.rb

```diff
@@ -8,6 +8,11 @@ class ApacheSpark < Formula
   license "Apache-2.0"
   head "https://github.com/apache/spark.git", branch: "master"
 
+  livecheck do
+    url "https://spark.apache.org/downloads.html"
+    regex(%r{>Spark v?(\d+(?:\.\d+)+) released}i)
```

Comment:
Looking at the download page, they create the download links using JavaScript. Checking the links in the "Latest News" sidebar (as you're doing) here is one option) and if we go that route, the regex could use some tweaks.

```suggestion
  livecheck do
    url "https://spark.apache.org/downloads.html"
    regex(%r{>\s*Spark\s+v?(\d+(?:\.\d+)+)\s+released\s*<}i)
```

However, if we want to check the versions that are used to create the download links on that page, we can check the `/js/downloads.js` file instead:

```suggestion
  # The download page creates file links using JavaScript, so we identify
  # versions within the related JS file.
  livecheck do
    url "https://spark.apache.org/js/downloads.js"
    regex(/addRelease\(.*?["']v?(\d+(?:\.\d+)+)["']/i)
```

They both work but there are benefits and detriments to each. If I had to choose...maybe the `download.js` approach, since it's what creates the links on the download page (i.e., there aren't new version links until that file is updated).